File: pro-lang/Java/Complex/src/Complex.java

public class Complex {
	public double imajener, reel;
	
	public Complex(double i, double r){
		this.imajener = i;
		this.reel = r;
	}
	public double GosterÝmajer(){
		return this.imajener;
	}
	public double Gosterreel(){
		return this.reel;
	}
	public String Goster(){
		return GosterÝmajer() + "i" + " + " + Gosterreel(); 
	}
	public Complex Topla(Complex other){
		return new Complex(this.imajener + other.imajener, this.reel + other.reel);
	}
	public Complex Cikar(Complex other){
		return new Complex(this.imajener - other.imajener, this.reel - other.reel);
	}
	public double Mutlak(){
		return Math.pow(imajener, 2) + Math.pow(imajener, 2);
	}
	public Complex Carpma(Complex other){
		double s1, s2;
		s1 = this.reel * other.reel + (-1 * this.imajener * other.imajener);
		s2 = this.imajener * other.reel + other.imajener * this.reel;
		return new Complex(s2, s1);
	}public Complex bolme(Complex other){
		double s1, s2, s3;
		s1 = this.reel * other.reel + (-1 * this.imajener * -1 * other.imajener);
		s2 = this.imajener * other.reel + other.imajener * this.reel;
		s3 = Math.pow(other.reel, 2) + (-1 * Math.pow(other.imajener, 2));
		return new Complex(s2 /s3, s1 / s3);
	}
	
	public static void main(String[] args) {
		// TODO Auto-generated method stub
		Complex nesne = new Complex(5.22, 7.9);
		Complex nesne1 = new Complex(4.1, -3.0);
		Complex sonuc = nesne.Topla(nesne1);
		Complex sonuc1 = nesne.Cikar(nesne1);
		Complex sonuc2 = nesne.Carpma(nesne1);
		Complex sonuc3 = nesne.bolme(nesne1);
		System.out.println("1.complex sayinin sanal veya imajener kismi " + nesne.GosterÝmajer());
		System.out.println("1.complex sayinin ree kismi " + nesne.Gosterreel());
		System.out.println("Toplama Sonuc : " + sonuc.Goster());
		System.out.println("Cýkarma sonuc : " + sonuc1.Goster());
		System.out.println("Carpma sonuc : " + sonuc2.Goster());
		System.out.println("Bolme Sonuc : " + sonuc3.Goster());
		System.out.println("complex sayinin mutlagý : " + nesne.Mutlak());
		System.out.println("1.complex sayi : " + nesne.Goster());
		System.out.println("2.complex sayini : " + nesne1.Goster());

	}

}

File: pro-lang/VeriYapilari/Complex.py
class Complex:
	
	def __init__(self, reel, sanal):
		self.re = reel
		self.imj = sanal
	
	def __str__(self):
		if self.re == 0:
			return str(self.imj) + "i"
		elif self.imj == 0:
			return str(self.re)
		else:
			return str(self.re) + "+" + str(self.imj) + "i"
	
	def show(self):
		print self.re, "+", self.sa, "i"
	
	def __toplam__(self, yeni):
		yeni_re = self.re + yeni.re
		yeni_imj = self.imj + yeni.imj
		return Complex(yeni_re, yeni_imj)
	
	def __cikarma__(self, yeni):
		yeni_re = self.re - yeni.re
		yeni_imj = self.imj - yeni.imj
		return Complex(yeni_re, yeni_imj)
	
	def __carpma__(self, yeni):
		if type(yeni) == int or type(yeni) == float:
			return Complex(self.re * yeni, self.imj * yeni)
		yeni_re = self.re * yeni.re + (-1) * self.imj * yeni.imj
		yeni_imj = self.re * yeni.imj + self.imj * yeni.re
		return Complex(yeni_re, yeni_imj)
	
	def __mutlak__(self, yeni):
		if type(yeni) == int or type(yeni) == float:
			return ((self.re - yeni) ** 2 + (self.imj) ** 2 ) ** 0.2
		return ((self.re - yeni.re) ** 2 + (self.imj - yeni.imj) ** 2) ** 0.2
	
	def __bolme__(self, yeni):
		payda = float((self.re ** 2) - (self.imj ** 2) * (-1))
		yeni_re = self.re * yeni.re + self.imj * yeni.imj
		yeni_imj = self.re * yeni.imj - self.imj * yeni.re 
		return Complex(yeni_re / payda, yeni_imj / payda)
f1 = Complex(1,2) 
f2 = Complex(1,0)
f3 = f1.__toplam__(f2)
f4 = f1.__cikarma__(f2)
f5 = f1.__carpma__(f2)
f6 = f1.__mutlak__(f2)
f7 = f1.__bolme__(f2)
print f3, f4 , f5, f6, f7
